app/src/main/java/com/example/counterapp/MainActivity.java

```diff
@@ -0,0 +1,95 @@
+package com.example.counterapp;
+
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.view.View;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.widget.Button;
+import android.widget.TextView;
+import android.widget.Toast;
+
+public class MainActivity extends AppCompatActivity {
+
+    //identifiers like txtCount are called member variables
+    //MainActivity is the class and an attribute or 'variable' of that class is TextView, which is a type of View. This particular TextView is named txtCount
+    TextView txtCount;
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+
+        txtCount= findViewById(R.id.counter_text);
+
+        //local variables can only be used within a specific method, unless referenced they cease to be usable after method is complete.
+        Button ResBtn=findViewById(R.id.res_btn);
+
+        ResBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                txtCount.setText(String.valueOf(0));
+                Toast.makeText(MainActivity.this, "resetting", Toast.LENGTH_SHORT).show();
+            }
+        });
+
+        Button SubBtn=findViewById(R.id.sub_btn);
+        SubBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String countValue=txtCount.getText().toString();
+                int intCountValue=Integer.parseInt(countValue);
+                intCountValue--;
+                txtCount.setText(String.valueOf(intCountValue));
+            }
+        });
+
+        Button AddBtn= findViewById(R.id.add_btn);
+        AddBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String countValue =txtCount.getText().toString();
+                int intCountValue =Integer.parseInt(countValue);
+                intCountValue++;
+                txtCount.setText(String.valueOf(intCountValue));
+            }
+        });
+
+    }
+
```

Comment:
I haven't yet taught you about standards for **method** and line level comments, but if you have some time, try Googling _"JavaDoc comment formatting"_. This will help you get a good understanding of why Java comments are the way they are. Also, you can see an example of how I might comment the `onCounterButtonClick` **method**:

```suggestion
    /**
     * A method that can be used as an action to take when clicking a counter button that will either increment 
     * or decrement the txtCount's value.
     * @param countingButton A reference to the button that was clicked and initiated this action
     * @param isIncrement A boolean flag to determine whether to increment or decrement the count value
     */
    private void onCounterButtonClick(Button countingButton, boolean isIncrement) {
```

Reply:
Do you want me to input your comment into the code? or was this just an example?

Reply:
It's best if it's written in your own words, but use the format I've given in my example. That includes the `@param` stuff as well. We'll talk more about documentation formatting and standards in one of our lessons.